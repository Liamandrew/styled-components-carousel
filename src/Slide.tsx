import styled from 'styled-components';

type Props = {
    width: number;
    key: string;
};

const Slide = styled.div.attrs<Props>(props => ({
    width: props.width || 0,
    key: props.key,
}))<Props>`
    display: block;
    width: ${props => props.width}px;
    outline: currentcolor none medium;
`;

export default Slide;
