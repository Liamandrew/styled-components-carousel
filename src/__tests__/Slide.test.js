import React from 'react';
import { render } from '@testing-library/react';
import Slide from '../Slide';

describe('Slide', () => {
    it('should match the snapshot when width is defined', () => {
        const { container } = render(<Slide width={300} />);

        expect(container).toMatchSnapshot();
    });

    it('should match the snapshot when width is undefined', () => {
        const { container } = render(<Slide />);

        expect(container).toMatchSnapshot();
    });
});
