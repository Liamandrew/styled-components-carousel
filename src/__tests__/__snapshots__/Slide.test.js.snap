// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should match the snapshot when width is defined 1`] = `
<div>
  <div
    class="sc-AxjAm epebAy"
    width="300"
  />
</div>
`;

exports[`Slide should match the snapshot when width is undefined 1`] = `
<div>
  <div
    class="sc-AxjAm cSilaZ"
  />
</div>
`;
